File: src/ShoppingCart.Business/Domain/Category.cs
using System;

namespace ShoppingCart.Business.Domain
{
    public class Category
    {
        public Guid Id { get; }

        public Guid? ParentCategoryID { get; }

        public string Title { get; }

        public Category(Guid id, Guid parentCategoryID, string title)
        {
            Id = id;
            ParentCategoryID = parentCategoryID;
            Title = title;
        }
    }
}

File: spring-boot/src/main/java/com/erdaldalkiran/cart/domain/category/Category.java
package com.erdaldalkiran.cart.domain.category;

import lombok.Getter;

import java.util.Optional;
import java.util.UUID;

@Getter
public class Category {
    private final UUID id;
    private final Optional<UUID> parentID;
    private final String title;

    public Category(UUID id, Optional<UUID> parentID, String title) {
        this.id = id;
        this.parentID = parentID;
        this.title = title;
    }
}
